packages/muelu/research/mmayr/composite_to_regions/src/SetupRegionSmoothers_def.hpp

```diff
@@ -265,26 +283,25 @@ void smootherSetup(RCP<Teuchos::ParameterList> params,
 {
   const std::string type = params->get<std::string>("smoother: type");
 
-  std::map<std::string, int> smootherTypes;
-  smootherTypes.insert(std::pair<std::string, int>("None",      0));
-  smootherTypes.insert(std::pair<std::string, int>("Jacobi",    1));
-  smootherTypes.insert(std::pair<std::string, int>("Gauss",     2));
-  smootherTypes.insert(std::pair<std::string, int>("Chebyshev", 3));
+  std::map<std::string, int> smootherTypes = getListOfValidSmootherTypes();
 
   switch(smootherTypes[type]) {
   case 0:
+  {
     break;
+  }
   case 1:
-    jacobiSetup(params, maxRegPerProc, revisedRowMapPerGrp, regionGrpMats, regionInterfaceScaling,
-                mapComp, rowMapPerGrp, rowImportPerGrp);
-    break;
   case 2:
-    jacobiSetup(params, maxRegPerProc, revisedRowMapPerGrp, regionGrpMats, regionInterfaceScaling,
+  {
+    relaxationSmootherSetup(params, maxRegPerProc, revisedRowMapPerGrp, regionGrpMats, regionInterfaceScaling,
                 mapComp, rowMapPerGrp, rowImportPerGrp);
     break;
+  }
   case 3:
-    std::cout << "Chebyshev smoother not implemented yet no smoother is applied" << std::endl;
+  {
+    std::cout << "Chebyshev smoother not implemented yet. No smoother is applied" << std::endl;
     break;
+  }
   default:
```

Comment:
Now that I look at this piece of code, I'm fairly convinced that we should add some brackets is we want the `throw` to work properly.

Reply:
@lucbv I added some `{`/`}`. Please re-approve.

Reply:
Thanks